project_earned_value_task/models/project.py

```diff
@@ -0,0 +1,472 @@
+
+        # CPI: Cost Performance Index
+        try:
+            res['cpi'] = res['ev'] / res['ac']
+        except ZeroDivisionError:
+            res['cpi'] = 1
+
+        # TCPI: To-complete Performance Index
+        bac_ac_amount = res['bac'] - res['ac']
+        try:
+            res['tcpi'] = (res['bac'] - res['ev']) / bac_ac_amount
+        except ZeroDivisionError:
+            res['tcpi'] = 1
+
+        # EAC: Estimate at completion
+        try:
+            res['eac'] = res['bac'] / res['cpi']
+        except ZeroDivisionError:
+            res['eac'] = res['bac']
+
+        # VAC: Variance at Completion
+        res['vac'] = res['bac'] - res['eac']
+
+        # VACP: Variance at Completion Percent
+        try:
+            res['vacp'] = (res['vac'] / res['bac']) * 100
+        except ZeroDivisionError:
+            res['vacp'] = 0
+
+        # ETC: Estimate To Complete
+        try:
+            res['etc'] = (res['bac'] - res['ev']) / res['cpi']
+        except ZeroDivisionError:
+            res['etc'] = 0
+
+        # EAC: Estimate At Completion
+        try:
+            res['eac'] = res['bac'] / res['cpi']
+        except ZeroDivisionError:
+            res['eac'] = 0
+
+        # POC - Percent of Completion
+        try:
+            res['poc'] = res['ev'] / res['bac'] * 100
+        except ZeroDivisionError:
+            res['poc'] = 0
+
+        return res
+
+    def _get_earliest_latest_dates(self):
+        project = self
+        cr = self.env.cr
+        # Get the earliest and latest dates for the associated tasks
+        cr.execute("""SELECT MIN(date_start)
+                   FROM project_task
+                   WHERE project_id=%s""", (project.id, ))
+        min_date_start = cr.fetchone()[0] or 0
+        if min_date_start == 0:
+            datetime_start = datetime.today()
+        else:
+            datetime_start = datetime.strptime(min_date_start,
+                                               "%Y-%m-%d %H:%M:%S")
+
+        cr.execute("""SELECT MAX(date_end)
+                   FROM project_task
+                   WHERE project_id=%s""", (project.id, ))
+        max_date_end = cr.fetchone()[0] or 0
+        if max_date_end == 0:
+            datetime_end = datetime.today()
+        else:
+            datetime_end = datetime.strptime(max_date_end,
+                                             "%Y-%m-%d %H:%M:%S")
+
+        cr.execute("""SELECT MAX(date_last_stage_update)
+                   FROM project_task
+                   WHERE project_id=%s""", (project.id, ))
+        max_date_end_2 = cr.fetchone()[0] or 0
+        if max_date_end_2 == 0:
+            datetime_end_2 = datetime.today()
+        else:
+            datetime_end_2 = datetime.strptime(max_date_end,
+                                             "%Y-%m-%d %H:%M:%S")
+        datetime_end = max(datetime_end, datetime_end_2)
+
+        return datetime_start, datetime_end
+
+    def _get_budget_at_completion(self):
```

Comment:
This is the same as `_get_plan_cost_to_date` using end date as measuring date. You should reuse that function.